ui/src/containers/organisations.list/serviceusers/tokens/add.tsx

```diff
@@ -0,0 +1,153 @@
+  const [isTokenLoading, setIsTokenLoading] = useState(false);
+
+  const onOpenChange = useCallback(() => {
+    navigate(`/organisations/${organisationId}/serviceusers/${serviceUserId}`);
+  }, [navigate, organisationId, serviceUserId]);
+
+  async function onSubmit(data: NewTokenBody) {
+    setIsTokenLoading(true);
+    try {
+      const resp = await client?.frontierServiceCreateServiceUserToken(
+        serviceUserId || "",
+        data
+      );
+
+      const generatedToken = resp?.data?.token;
+      setToken(generatedToken);
+    } catch (err: any) {
+      console.error(err);
+      toast.error("Unable to create service user token", err);
+    } finally {
+      setIsTokenLoading(false);
+    }
+  }
+
+  const methods = useForm<NewTokenBody>({
+    resolver: zodResolver(NewTokenBodySchema),
+    defaultValues: {},
+  });
+
+  return (
+    <Sheet open={true}>
+      <Sheet.Content
+        side="right"
+        // @ts-ignore
+        style={{
+          width: "30vw",
+          padding: 0,
+          borderRadius: "var(--pd-8)",
+          boxShadow: "var(--shadow-sm)",
+        }}
+        close={false}
+      >
+        <FormProvider {...methods}>
+          <Form onSubmit={methods.handleSubmit(onSubmit)}>
+            <SheetHeader
+              title={"Generate Token"}
+              onClick={onOpenChange}
+            ></SheetHeader>
+            <Flex
+              style={{ padding: "24px" }}
+              direction={"column"}
+              gap={"medium"}
+            >
+              <CustomFieldName
+                name="title"
+                register={methods.register}
+                control={methods.control}
+              />
+              <Separator />
+
+              {isTokenLoading ? (
+                <Skeleton
+                  style={{
+                    height: "36px",
+                  }}
+                />
+              ) : token?.id ? (
+                <Flex direction={"column"} gap={"small"}>
+                  <Label size={"medium"}>ID</Label>
+                  <Text size={3} className={styles.tokenTextDiv}>
+                    {token?.id}
+                  </Text>
+                </Flex>
+              ) : null}
+
+              {isTokenLoading ? (
+                <Skeleton
+                  style={{
+                    height: "36px",
+                  }}
+                />
+              ) : token?.token ? (
+                <Flex direction={"column"} gap={"small"}>
+                  <Label size={"medium"}>Token</Label>
+                  <Text size={3} className={styles.tokenTextDiv}>
+                    {token?.token}
+                  </Text>
+                </Flex>
+              ) : null}
+
+              {token?.token ? (
+                <Text
+                  size={4}
+                  weight={500}
+                  style={{ color: "var(--foreground-danger)" }}
+                >
+                  For safety reasons, we can't show it again. Please copy it
```

Comment:
What is `it` here? Can we explicitly use the word `the token` here?

Reply:
not just token, user has to copy id as well.

Reply:
In that case, can we mention it as "the credentials" instead of "it"?
Case in point: I thought it meant only the secret, not Id. Someone else might also misinterpret the text.